### PR TITLE
MoveTables: add flag to specify that routing rules should not be created when a movetables workflow is created

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description

This PR adds a new flag `--no-routing-rules` to MoveTables. Usually when a MoveTables workflow is created we create routing rules pointing the participating tables to the source keyspace and then reroute them to the target keyspace after traffic is switched.

However for some use cases, like specified in #13851, we do not want these routing rules to be created. The expected flow for workflows using this routing rule is:
* taking the app offline
* waiting manually for the workflow to catchup with the source
* `cancel` the workflow with `--keep-data` 
* update the app config to point to the target
* bring the app back online

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/13851
  
Docs: https://github.com/vitessio/website/pull/1573

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [X] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [X] Documentation was added or is not required

### CODE REVIEW

**Review by dbussink — APPROVED** [review]

File: go/cmd/vtctldclient/command/movetables.go [hunk 1]
```diff
@@ -524,6 +526,7 @@ func init() {
 	MoveTablesCreate.Flags().BoolVar(&moveTablesCreateOptions.DeferSecondaryKeys, "defer-secondary-keys", false, "Defer secondary index creation for a table until after it has been copied")
 	MoveTablesCreate.Flags().BoolVar(&moveTablesCreateOptions.AutoStart, "auto-start", true, "Start the MoveTables workflow after creating it")
 	MoveTablesCreate.Flags().BoolVar(&moveTablesCreateOptions.StopAfterCopy, "stop-after-copy", false, "Stop the MoveTables workflow after it's finished copying the existing rows and before it starts replicating changes")
+	MoveTablesCreate.Flags().BoolVar(&moveTablesCreateOptions.NoRoutingRules, "no-routing-rules", false, "(Advanced) MoveTables Create only. Do not create routing rules while creating the workflow. See the reference documentation for limitations if you use this flag.")
```
Comment (mattlord) [inline go/cmd/vtctldclient/command/movetables.go:529]:
```suggestion
	MoveTablesCreate.Flags().BoolVar(&moveTablesCreateOptions.NoRoutingRules, "no-routing-rules", false, "(Advanced) Do not create routing rules while creating the workflow. See the reference documentation for limitations if you use this flag.")
```

**Review by mattlord — COMMENTED** [review]
Looks good! IMO we should add a unit test of some kind on the vtctldclient side too. 

We also create/update routing rules when we switch traffic... but it doesn't seem like we take this flag into account then. Shouldn't we do so then as well?

**Comment by rohit-nayak-ps** [comment]
> Looks good! IMO we should add a unit test of some kind on the vtctldclient side too.
Already added `TestMoveTablesNoRoutingRules`: https://github.com/vitessio/vitess/pull/13895/files#diff-d9ebea8e2869790fa81e2a78fa3e899cfd220b9353ae0a7ad6d12c472250a89eR526

> We also create/update routing rules when we switch traffic... but it doesn't seem like we take this flag into account then. Shouldn't we do so then as well?

Yeah, however today creating the rules after switching traffic is necessary. It allows the existing switching functionality to work, since we look at the rules to decide what traffic is switched and print the current state of the workflow. Otherwise we will need to store the workflow state separately, in itself a non-trivial design change since we will now have state in two different places, the need to reconcile them, etc  ...

Maybe the flag could be differently named? However since it is a MoveTables Create-only flag, we can argue that this is while creating the workflow and not in SwitchTraffic.

**Comment by mattlord** [comment]
> > Looks good! IMO we should add a unit test of some kind on the vtctldclient side too.
> Already added `TestMoveTablesNoRoutingRules`: https://github.com/vitessio/vitess/pull/13895/files#diff-d9ebea8e2869790fa81e2a78fa3e899cfd220b9353ae0a7ad6d12c472250a89eR526

Ah! Looks like somebody needs new glasses. 😄 
 
> > We also create/update routing rules when we switch traffic... but it doesn't seem like we take this flag into account then. Shouldn't we do so then as well?
> 
> Yeah, however today creating the rules after switching traffic is necessary. It allows the existing switching functionality to work, since we look at the rules to decide what traffic is switched and print the current state of the workflow. Otherwise we will need to store the workflow state separately, in itself a non-trivial design change since we will now have state in two different places, the need to reconcile them, etc ...
> 
> Maybe the flag could be differently named? However since it is a MoveTables Create-only flag, we can argue that this is while creating the workflow and not in SwitchTraffic.

👍

**Review by mattlord — APPROVED** [review]

**Pull request merged**